### PR TITLE
[web] Fix semantic node order for webkit

### PR DESCRIPTION
**Pull request by ferhatb**
## Description

Voiceover fails to order elements if transform, left, top is not specified on semantics nodes. This PR disables the optimization to skip translation styles when matrix is identity and offset is 0. 

PR also removes semantics test skips for Edge due to style order.

## Related Issues

https://github.com/flutter/flutter/issues/73347
https://github.com/flutter/flutter/issues/50754/

## Tests

I added the following tests:

semantics_test.dart

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [x] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

### CODE REVIEW

File: lib/web_ui/lib/src/engine/semantics/semantics.dart [hunk 1]
```diff
@@ -849,30 +849,14 @@ class SemanticsObject {
     final bool hasIdentityTransform =
         transform == null || isIdentityFloat32ListTransform(transform);
 
+    // On Mac OS and iOS, VoiceOver requires left=0 top=0 value to correctly
+    // handle order. See https://github.com/flutter/flutter/issues/73347.
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/semantics/semantics.dart:853]: Should this comment be closer to the `_resetContainerOffsets` and `_resetElementOffsets` functions?

File: lib/web_ui/lib/src/engine/semantics/semantics.dart [hunk 2]
```diff
@@ -968,6 +969,30 @@ class SemanticsObject {
...
+  void _resetElementOffsets() {
+    if (isMacOrIOS) {
+      if (isDesktop) {
+        element.style
+          ..transformOrigin = '0 0 0'
+          ..transform = 'translate(0px, 0px)';
+      } else {
+        element.style
+          ..top = '0px'
+          ..left = '0px';
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/semantics/semantics.dart:981]: What happens if we use top/left in all browsers?

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

File: lib/web_ui/lib/src/engine/semantics/semantics.dart (hunk 2 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/semantics/semantics.dart:981]: The idea was that layout faster with transform.

File: lib/web_ui/lib/src/engine/semantics/semantics.dart (hunk 1 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/semantics/semantics.dart]: done

File: lib/web_ui/lib/src/engine/semantics/semantics.dart [hunk 3]
```diff
@@ -853,26 +853,8 @@ class SemanticsObject {
...
-            ..removeProperty('transform-origin')
-            ..removeProperty('transform');
-        } else {
-          containerElement.style
-            ..removeProperty('top')
-            ..removeProperty('left');
-        }
-      }
+      _resetElementOffsets();
+      _resetContainerOffsets(containerElement);
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/semantics/semantics.dart:857]: It looks like these two methods do the exact same thing but on different html elements. Maybe consolidate them into one method that takes the element as an argument?

Comment (ferhatb) [reply lib/web_ui/lib/src/engine/semantics/semantics.dart]: Thanks. Done.
